src/host/outputStream.cpp

```diff
@@ -893,6 +893,30 @@ void ConhostInternalGetSet::UpdateSoftFont(const gsl::span<const uint16_t> bitPa
     }
 }
 
+// Method Description:
+// - Inform the console that the window is focused. This is used by ConPTY.
+//   Terminals can send ConPTY a FocusIn/FocusOut sequence on the input pipe,
+//   which will end up here. This will update the console's internal tracker if
+//   it's focused or not, as to match the end-terminal's state.
+// - Used to call ConsoleControl(ConsoleSetForeground,...).
+// - Full support for this sequence is tracked in GH#11682.
+// Arguments:
+// - bitPattern - An array of scanlines representing all the glyphs in the font.
+// - cellSize - The cell size for an individual glyph.
+// - centeringHint - The horizontal extent that glyphs are offset from center.
+// Return Value:
+// - <none>
+void ConhostInternalGetSet::FocusChanged(const bool focused)
+{
+    auto& g = ServiceLocator::LocateGlobals();
+    auto& gci = g.getConsoleInformation();
+    WI_UpdateFlag(gci.Flags, CONSOLE_HAS_FOCUS, focused);
+    gci.ProcessHandleList.ModifyConsoleProcessFocus(focused);
+
+    // Theoretically, this could be propagated as a focus event as well, to the
+    // input buffer. That should be considered when implementing GH#11682.
+}

```

Comment:
If this isn't called from `AdaptDispatch` (which doesn't seem to be the case), then it shouldn't be part of the `ConGetSet` api. I would assume this code could be included directly in the `InteractDispatch` method.

Reply:
Good catch, glad to get that simplified